### PR TITLE
Introduce a natural compare func for sorting array and using it for rinv/rvital

### LINKED ISSUE #4298: [OpenBMC] Sorting code is removing zeros from string
Found in Austin Cluster: https://w3.rchland.ibm.com/projects/bestquest/?defect=SW405512

```
rinv cn05 serial 
cn05: SYSTEM SerialNumber : 781A9A
```
The serial number is actually `781A09A`

I can re-create it on our machine if I replace the `$content_info` with that number: 

```
rinv mid05tor12cn11 serial
....
Tue Nov 14 14:55:08 2017 mid05tor12cn11: [openbmc_debug] 781A09A
mid05tor12cn11: 781A9A
```

I'm not sure if this only affects serial number, but i think it should affect all code that runs through this sorting..

### PR DESCRIPTION
**Pull request by robin2008**
To fix #4298 

1， add a new compare func in xCAT::Utils
2,   in openbmc plugin to use it for sorting the output by natural.

Note:  it did not cover other place to use the natural sorting 

UT:
```
[root@briggs01 xcat]# rinv mid05tor12cn05 all
mid05tor12cn05: BMC BuildDate :
mid05tor12cn05: BMC FieldReplaceable : 0
mid05tor12cn05: BMC Manufacturer : IBM
mid05tor12cn05: BMC Model :
mid05tor12cn05: BMC PartNumber : 01DH118
mid05tor12cn05: BMC Present : 1
mid05tor12cn05: BMC PrettyName : BMC PLANAR
mid05tor12cn05: BMC SerialNumber : 000000000000
mid05tor12cn05: BMC UUID :
mid05tor12cn05: CPU0 BuildDate : 1996-01-01 - 00:00:00
mid05tor12cn05: CPU0 CORE0 Functional : 1
mid05tor12cn05: CPU0 CORE0 Present : 0
mid05tor12cn05: CPU0 CORE0 PrettyName :
mid05tor12cn05: CPU0 CORE1 Functional : 1
mid05tor12cn05: CPU0 CORE1 Present : 0
mid05tor12cn05: CPU0 CORE1 PrettyName :
mid05tor12cn05: CPU0 CORE2 Functional : 1
mid05tor12cn05: CPU0 CORE2 Present : 0
mid05tor12cn05: CPU0 CORE2 PrettyName :
mid05tor12cn05: CPU0 CORE3 Functional : 1
mid05tor12cn05: CPU0 CORE3 Present : 0
mid05tor12cn05: CPU0 CORE3 PrettyName :
mid05tor12cn05: CPU0 CORE4 Functional : 1
mid05tor12cn05: CPU0 CORE4 Present : 1
mid05tor12cn05: CPU0 CORE4 PrettyName :
mid05tor12cn05: CPU0 CORE5 Functional : 1
mid05tor12cn05: CPU0 CORE5 Present : 1
mid05tor12cn05: CPU0 CORE5 PrettyName :
mid05tor12cn05: CPU0 CORE6 Functional : 1
mid05tor12cn05: CPU0 CORE6 Present : 0
mid05tor12cn05: CPU0 CORE6 PrettyName :
mid05tor12cn05: CPU0 CORE7 Functional : 1
mid05tor12cn05: CPU0 CORE7 Present : 0
mid05tor12cn05: CPU0 CORE7 PrettyName :
mid05tor12cn05: CPU0 CORE8 Functional : 1
mid05tor12cn05: CPU0 CORE8 Present : 1
mid05tor12cn05: CPU0 CORE8 PrettyName :
mid05tor12cn05: CPU0 CORE9 Functional : 1
mid05tor12cn05: CPU0 CORE9 Present : 1
mid05tor12cn05: CPU0 CORE9 PrettyName :
mid05tor12cn05: CPU0 CORE10 Functional : 1
mid05tor12cn05: CPU0 CORE10 Present : 1
mid05tor12cn05: CPU0 CORE10 PrettyName :
mid05tor12cn05: CPU0 CORE11 Functional : 1
mid05tor12cn05: CPU0 CORE11 Present : 1
mid05tor12cn05: CPU0 CORE11 PrettyName :
mid05tor12cn05: CPU0 CORE12 Functional : 1
mid05tor12cn05: CPU0 CORE12 Present : 1
mid05tor12cn05: CPU0 CORE12 PrettyName :
mid05tor12cn05: CPU0 CORE13 Functional : 1
mid05tor12cn05: CPU0 CORE13 Present : 1
mid05tor12cn05: CPU0 CORE13 PrettyName :
mid05tor12cn05: CPU0 CORE14 Functional : 1
mid05tor12cn05: CPU0 CORE14 Present : 1
mid05tor12cn05: CPU0 CORE14 PrettyName :
mid05tor12cn05: CPU0 CORE15 Functional : 1
mid05tor12cn05: CPU0 CORE15 Present : 1
mid05tor12cn05: CPU0 CORE15 PrettyName :
mid05tor12cn05: CPU0 CORE16 Functional : 1
mid05tor12cn05: CPU0 CORE16 Present : 1
mid05tor12cn05: CPU0 CORE16 PrettyName :
mid05tor12cn05: CPU0 CORE17 Functional : 1
mid05tor12cn05: CPU0 CORE17 Present : 1
mid05tor12cn05: CPU0 CORE17 PrettyName :
mid05tor12cn05: CPU0 CORE18 Functional : 1
mid05tor12cn05: CPU0 CORE18 Present : 1
mid05tor12cn05: CPU0 CORE18 PrettyName :
mid05tor12cn05: CPU0 CORE19 Functional : 1
mid05tor12cn05: CPU0 CORE19 Present : 1
mid05tor12cn05: CPU0 CORE19 PrettyName :
mid05tor12cn05: CPU0 CORE20 Functional : 1
mid05tor12cn05: CPU0 CORE20 Present : 1
mid05tor12cn05: CPU0 CORE20 PrettyName :
mid05tor12cn05: CPU0 CORE21 Functional : 1
mid05tor12cn05: CPU0 CORE21 Present : 1
mid05tor12cn05: CPU0 CORE21 PrettyName :
mid05tor12cn05: CPU0 CORE22 Functional : 1
mid05tor12cn05: CPU0 CORE22 Present : 1
mid05tor12cn05: CPU0 CORE22 PrettyName :
mid05tor12cn05: CPU0 CORE23 Functional : 1
mid05tor12cn05: CPU0 CORE23 Present : 1
mid05tor12cn05: CPU0 CORE23 PrettyName :
mid05tor12cn05: CPU0 Cached : 0
mid05tor12cn05: CPU0 FieldReplaceable : 1
mid05tor12cn05: CPU0 Functional : 1
mid05tor12cn05: CPU0 Manufacturer : IBM
mid05tor12cn05: CPU0 Model :
mid05tor12cn05: CPU0 PartNumber : 02AA766
mid05tor12cn05: CPU0 Present : 1
mid05tor12cn05: CPU0 PrettyName : PROCESSOR MODULE
mid05tor12cn05: CPU0 SerialNumber : YA1934017985
mid05tor12cn05: CPU0 Version : 20
mid05tor12cn05: CPU1 BuildDate : 1996-01-01 - 00:00:00
mid05tor12cn05: CPU1 CORE0 Functional : 1
mid05tor12cn05: CPU1 CORE0 Present : 0
mid05tor12cn05: CPU1 CORE0 PrettyName :
mid05tor12cn05: CPU1 CORE1 Functional : 1
mid05tor12cn05: CPU1 CORE1 Present : 0
mid05tor12cn05: CPU1 CORE1 PrettyName :
mid05tor12cn05: CPU1 CORE2 Functional : 1
mid05tor12cn05: CPU1 CORE2 Present : 0
mid05tor12cn05: CPU1 CORE2 PrettyName :
mid05tor12cn05: CPU1 CORE3 Functional : 1
mid05tor12cn05: CPU1 CORE3 Present : 0
mid05tor12cn05: CPU1 CORE3 PrettyName :
mid05tor12cn05: CPU1 CORE4 Functional : 1
mid05tor12cn05: CPU1 CORE4 Present : 1
mid05tor12cn05: CPU1 CORE4 PrettyName :
mid05tor12cn05: CPU1 CORE5 Functional : 1
mid05tor12cn05: CPU1 CORE5 Present : 1
mid05tor12cn05: CPU1 CORE5 PrettyName :
mid05tor12cn05: CPU1 CORE6 Functional : 1
mid05tor12cn05: CPU1 CORE6 Present : 1
mid05tor12cn05: CPU1 CORE6 PrettyName :
mid05tor12cn05: CPU1 CORE7 Functional : 1
mid05tor12cn05: CPU1 CORE7 Present : 1
mid05tor12cn05: CPU1 CORE7 PrettyName :
mid05tor12cn05: CPU1 CORE8 Functional : 1
mid05tor12cn05: CPU1 CORE8 Present : 1
mid05tor12cn05: CPU1 CORE8 PrettyName :
mid05tor12cn05: CPU1 CORE9 Functional : 1
mid05tor12cn05: CPU1 CORE9 Present : 1
mid05tor12cn05: CPU1 CORE9 PrettyName :
mid05tor12cn05: CPU1 CORE10 Functional : 1
mid05tor12cn05: CPU1 CORE10 Present : 1
mid05tor12cn05: CPU1 CORE10 PrettyName :
mid05tor12cn05: CPU1 CORE11 Functional : 1
mid05tor12cn05: CPU1 CORE11 Present : 1
mid05tor12cn05: CPU1 CORE11 PrettyName :
mid05tor12cn05: CPU1 CORE12 Functional : 1
mid05tor12cn05: CPU1 CORE12 Present : 1
mid05tor12cn05: CPU1 CORE12 PrettyName :
mid05tor12cn05: CPU1 CORE13 Functional : 1
mid05tor12cn05: CPU1 CORE13 Present : 1
mid05tor12cn05: CPU1 CORE13 PrettyName :
mid05tor12cn05: CPU1 CORE14 Functional : 1
mid05tor12cn05: CPU1 CORE14 Present : 1
mid05tor12cn05: CPU1 CORE14 PrettyName :
mid05tor12cn05: CPU1 CORE15 Functional : 1
mid05tor12cn05: CPU1 CORE15 Present : 1
mid05tor12cn05: CPU1 CORE15 PrettyName :
mid05tor12cn05: CPU1 CORE16 Functional : 1
mid05tor12cn05: CPU1 CORE16 Present : 1
mid05tor12cn05: CPU1 CORE16 PrettyName :
mid05tor12cn05: CPU1 CORE17 Functional : 1
mid05tor12cn05: CPU1 CORE17 Present : 1
mid05tor12cn05: CPU1 CORE17 PrettyName :
mid05tor12cn05: CPU1 CORE18 Functional : 1
mid05tor12cn05: CPU1 CORE18 Present : 1
mid05tor12cn05: CPU1 CORE18 PrettyName :
mid05tor12cn05: CPU1 CORE19 Functional : 1
mid05tor12cn05: CPU1 CORE19 Present : 1
mid05tor12cn05: CPU1 CORE19 PrettyName :
mid05tor12cn05: CPU1 CORE20 Functional : 1
mid05tor12cn05: CPU1 CORE20 Present : 0
mid05tor12cn05: CPU1 CORE20 PrettyName :
mid05tor12cn05: CPU1 CORE21 Functional : 1
mid05tor12cn05: CPU1 CORE21 Present : 0
mid05tor12cn05: CPU1 CORE21 PrettyName :
mid05tor12cn05: CPU1 CORE22 Functional : 1
mid05tor12cn05: CPU1 CORE22 Present : 1
mid05tor12cn05: CPU1 CORE22 PrettyName :
mid05tor12cn05: CPU1 CORE23 Functional : 1
mid05tor12cn05: CPU1 CORE23 Present : 1
mid05tor12cn05: CPU1 CORE23 PrettyName :
mid05tor12cn05: CPU1 Cached : 0
mid05tor12cn05: CPU1 FieldReplaceable : 1
mid05tor12cn05: CPU1 Functional : 1
mid05tor12cn05: CPU1 Manufacturer : IBM
mid05tor12cn05: CPU1 Model :
mid05tor12cn05: CPU1 PartNumber : 02AA766
mid05tor12cn05: CPU1 Present : 1
mid05tor12cn05: CPU1 PrettyName : PROCESSOR MODULE
mid05tor12cn05: CPU1 SerialNumber : YA1934017988
mid05tor12cn05: CPU1 Version : 20
mid05tor12cn05: DIMM0 BuildDate :
mid05tor12cn05: DIMM0 Cached : 0
mid05tor12cn05: DIMM0 FieldReplaceable : 1
mid05tor12cn05: DIMM0 Functional : 1
mid05tor12cn05: DIMM0 Manufacturer : 0x2c80
mid05tor12cn05: DIMM0 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM0 PartNumber :
mid05tor12cn05: DIMM0 Present : 1
mid05tor12cn05: DIMM0 PrettyName : 0x0c
mid05tor12cn05: DIMM0 SerialNumber : 0x1611e800
mid05tor12cn05: DIMM0 Version : 0x31
mid05tor12cn05: DIMM1 BuildDate :
mid05tor12cn05: DIMM1 Cached : 0
mid05tor12cn05: DIMM1 FieldReplaceable : 1
mid05tor12cn05: DIMM1 Functional : 1
mid05tor12cn05: DIMM1 Manufacturer : 0x2c80
mid05tor12cn05: DIMM1 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM1 PartNumber :
mid05tor12cn05: DIMM1 Present : 1
mid05tor12cn05: DIMM1 PrettyName : 0x0c
mid05tor12cn05: DIMM1 SerialNumber : 0x1611e82d
mid05tor12cn05: DIMM1 Version : 0x31
mid05tor12cn05: DIMM2 BuildDate :
mid05tor12cn05: DIMM2 Cached : 0
mid05tor12cn05: DIMM2 FieldReplaceable : 1
mid05tor12cn05: DIMM2 Functional : 1
mid05tor12cn05: DIMM2 Manufacturer : 0x2c80
mid05tor12cn05: DIMM2 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM2 PartNumber :
mid05tor12cn05: DIMM2 Present : 1
mid05tor12cn05: DIMM2 PrettyName : 0x0c
mid05tor12cn05: DIMM2 SerialNumber : 0x1611e538
mid05tor12cn05: DIMM2 Version : 0x31
mid05tor12cn05: DIMM3 BuildDate :
mid05tor12cn05: DIMM3 Cached : 0
mid05tor12cn05: DIMM3 FieldReplaceable : 1
mid05tor12cn05: DIMM3 Functional : 1
mid05tor12cn05: DIMM3 Manufacturer : 0x2c80
mid05tor12cn05: DIMM3 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM3 PartNumber :
mid05tor12cn05: DIMM3 Present : 1
mid05tor12cn05: DIMM3 PrettyName : 0x0c
mid05tor12cn05: DIMM3 SerialNumber : 0x1611e837
mid05tor12cn05: DIMM3 Version : 0x31
mid05tor12cn05: DIMM4 BuildDate :
mid05tor12cn05: DIMM4 Cached : 0
mid05tor12cn05: DIMM4 FieldReplaceable : 1
mid05tor12cn05: DIMM4 Functional : 1
mid05tor12cn05: DIMM4 Manufacturer : 0x2c80
mid05tor12cn05: DIMM4 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM4 PartNumber :
mid05tor12cn05: DIMM4 Present : 1
mid05tor12cn05: DIMM4 PrettyName : 0x0c
mid05tor12cn05: DIMM4 SerialNumber : 0x1610acde
mid05tor12cn05: DIMM4 Version : 0x31
mid05tor12cn05: DIMM5 BuildDate :
mid05tor12cn05: DIMM5 Cached : 0
mid05tor12cn05: DIMM5 FieldReplaceable : 1
mid05tor12cn05: DIMM5 Functional : 1
mid05tor12cn05: DIMM5 Manufacturer : 0x2c80
mid05tor12cn05: DIMM5 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM5 PartNumber :
mid05tor12cn05: DIMM5 Present : 1
mid05tor12cn05: DIMM5 PrettyName : 0x0c
mid05tor12cn05: DIMM5 SerialNumber : 0x1610ace7
mid05tor12cn05: DIMM5 Version : 0x31
mid05tor12cn05: DIMM6 BuildDate :
mid05tor12cn05: DIMM6 Cached : 0
mid05tor12cn05: DIMM6 FieldReplaceable : 1
mid05tor12cn05: DIMM6 Functional : 1
mid05tor12cn05: DIMM6 Manufacturer : 0x2c80
mid05tor12cn05: DIMM6 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM6 PartNumber :
mid05tor12cn05: DIMM6 Present : 1
mid05tor12cn05: DIMM6 PrettyName : 0x0c
mid05tor12cn05: DIMM6 SerialNumber : 0x1610ad0c
mid05tor12cn05: DIMM6 Version : 0x31
mid05tor12cn05: DIMM7 BuildDate :
mid05tor12cn05: DIMM7 Cached : 0
mid05tor12cn05: DIMM7 FieldReplaceable : 1
mid05tor12cn05: DIMM7 Functional : 1
mid05tor12cn05: DIMM7 Manufacturer : 0x2c80
mid05tor12cn05: DIMM7 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM7 PartNumber :
mid05tor12cn05: DIMM7 Present : 1
mid05tor12cn05: DIMM7 PrettyName : 0x0c
mid05tor12cn05: DIMM7 SerialNumber : 0x1610acc0
mid05tor12cn05: DIMM7 Version : 0x31
mid05tor12cn05: DIMM8 BuildDate :
mid05tor12cn05: DIMM8 Cached : 0
mid05tor12cn05: DIMM8 FieldReplaceable : 1
mid05tor12cn05: DIMM8 Functional : 1
mid05tor12cn05: DIMM8 Manufacturer : 0x2c80
mid05tor12cn05: DIMM8 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM8 PartNumber :
mid05tor12cn05: DIMM8 Present : 1
mid05tor12cn05: DIMM8 PrettyName : 0x0c
mid05tor12cn05: DIMM8 SerialNumber : 0x1610ad8b
mid05tor12cn05: DIMM8 Version : 0x31
mid05tor12cn05: DIMM9 BuildDate :
mid05tor12cn05: DIMM9 Cached : 0
mid05tor12cn05: DIMM9 FieldReplaceable : 1
mid05tor12cn05: DIMM9 Functional : 1
mid05tor12cn05: DIMM9 Manufacturer : 0x2c80
mid05tor12cn05: DIMM9 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM9 PartNumber :
mid05tor12cn05: DIMM9 Present : 1
mid05tor12cn05: DIMM9 PrettyName : 0x0c
mid05tor12cn05: DIMM9 SerialNumber : 0x1610acc4
mid05tor12cn05: DIMM9 Version : 0x31
mid05tor12cn05: DIMM10 BuildDate :
mid05tor12cn05: DIMM10 Cached : 0
mid05tor12cn05: DIMM10 FieldReplaceable : 1
mid05tor12cn05: DIMM10 Functional : 1
mid05tor12cn05: DIMM10 Manufacturer : 0x2c80
mid05tor12cn05: DIMM10 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM10 PartNumber :
mid05tor12cn05: DIMM10 Present : 1
mid05tor12cn05: DIMM10 PrettyName : 0x0c
mid05tor12cn05: DIMM10 SerialNumber : 0x1610acd7
mid05tor12cn05: DIMM10 Version : 0x31
mid05tor12cn05: DIMM11 BuildDate :
mid05tor12cn05: DIMM11 Cached : 0
mid05tor12cn05: DIMM11 FieldReplaceable : 1
mid05tor12cn05: DIMM11 Functional : 1
mid05tor12cn05: DIMM11 Manufacturer : 0x2c80
mid05tor12cn05: DIMM11 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM11 PartNumber :
mid05tor12cn05: DIMM11 Present : 1
mid05tor12cn05: DIMM11 PrettyName : 0x0c
mid05tor12cn05: DIMM11 SerialNumber : 0x1610ace4
mid05tor12cn05: DIMM11 Version : 0x31
mid05tor12cn05: DIMM12 BuildDate :
mid05tor12cn05: DIMM12 Cached : 0
mid05tor12cn05: DIMM12 FieldReplaceable : 1
mid05tor12cn05: DIMM12 Functional : 1
mid05tor12cn05: DIMM12 Manufacturer : 0x2c80
mid05tor12cn05: DIMM12 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM12 PartNumber :
mid05tor12cn05: DIMM12 Present : 1
mid05tor12cn05: DIMM12 PrettyName : 0x0c
mid05tor12cn05: DIMM12 SerialNumber : 0x1610ad72
mid05tor12cn05: DIMM12 Version : 0x31
mid05tor12cn05: DIMM13 BuildDate :
mid05tor12cn05: DIMM13 Cached : 0
mid05tor12cn05: DIMM13 FieldReplaceable : 1
mid05tor12cn05: DIMM13 Functional : 1
mid05tor12cn05: DIMM13 Manufacturer : 0x2c80
mid05tor12cn05: DIMM13 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM13 PartNumber :
mid05tor12cn05: DIMM13 Present : 1
mid05tor12cn05: DIMM13 PrettyName : 0x0c
mid05tor12cn05: DIMM13 SerialNumber : 0x1610ab82
mid05tor12cn05: DIMM13 Version : 0x31
mid05tor12cn05: DIMM14 BuildDate :
mid05tor12cn05: DIMM14 Cached : 0
mid05tor12cn05: DIMM14 FieldReplaceable : 1
mid05tor12cn05: DIMM14 Functional : 1
mid05tor12cn05: DIMM14 Manufacturer : 0x2c80
mid05tor12cn05: DIMM14 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM14 PartNumber :
mid05tor12cn05: DIMM14 Present : 1
mid05tor12cn05: DIMM14 PrettyName : 0x0c
mid05tor12cn05: DIMM14 SerialNumber : 0x1610a5b8
mid05tor12cn05: DIMM14 Version : 0x31
mid05tor12cn05: DIMM15 BuildDate :
mid05tor12cn05: DIMM15 Cached : 0
mid05tor12cn05: DIMM15 FieldReplaceable : 1
mid05tor12cn05: DIMM15 Functional : 1
mid05tor12cn05: DIMM15 Manufacturer : 0x2c80
mid05tor12cn05: DIMM15 Model : 36ASF4G72PZ-2G6D1
mid05tor12cn05: DIMM15 PartNumber :
mid05tor12cn05: DIMM15 Present : 1
mid05tor12cn05: DIMM15 PrettyName : 0x0c
mid05tor12cn05: DIMM15 SerialNumber : 0x1610a632
mid05tor12cn05: DIMM15 Version : 0x31
mid05tor12cn05: ETHERNET FieldReplaceable : 0
mid05tor12cn05: ETHERNET MACAddress : 70:e2:84:14:28:0d
mid05tor12cn05: ETHERNET Present : 1
mid05tor12cn05: ETHERNET PrettyName :
mid05tor12cn05: FAN0 Functional : 1
mid05tor12cn05: FAN0 MeetsMinimumShipLevel : 1
mid05tor12cn05: FAN0 Present : 1
mid05tor12cn05: FAN0 PrettyName : fan0
mid05tor12cn05: FAN1 Functional : 1
mid05tor12cn05: FAN1 MeetsMinimumShipLevel : 1
mid05tor12cn05: FAN1 Present : 1
mid05tor12cn05: FAN1 PrettyName : fan1
mid05tor12cn05: FAN2 Functional : 1
mid05tor12cn05: FAN2 MeetsMinimumShipLevel : 1
mid05tor12cn05: FAN2 Present : 1
mid05tor12cn05: FAN2 PrettyName : fan2
mid05tor12cn05: FAN3 Functional : 1
mid05tor12cn05: FAN3 MeetsMinimumShipLevel : 1
mid05tor12cn05: FAN3 Present : 1
mid05tor12cn05: FAN3 PrettyName : fan3
mid05tor12cn05: MOTHERBOARD BuildDate : 1996-01-01 - 00:00:00
mid05tor12cn05: MOTHERBOARD Cached : 0
mid05tor12cn05: MOTHERBOARD FieldReplaceable : 0
mid05tor12cn05: MOTHERBOARD Manufacturer : 0000000000000000
mid05tor12cn05: MOTHERBOARD Model :
mid05tor12cn05: MOTHERBOARD PartNumber : 00VK614
mid05tor12cn05: MOTHERBOARD Present : 1
mid05tor12cn05: MOTHERBOARD PrettyName : SYSTEM PLANAR
mid05tor12cn05: MOTHERBOARD SerialNumber : YL30UF75901T
mid05tor12cn05: POWER_SUPPLY0 Present : 1
mid05tor12cn05: POWER_SUPPLY0 PrettyName : power_supply0
mid05tor12cn05: POWER_SUPPLY1 Present : 1
mid05tor12cn05: POWER_SUPPLY1 PrettyName : power_supply1
mid05tor12cn05: SYSTEM AssetTag :
mid05tor12cn05: SYSTEM BuildDate :
mid05tor12cn05: SYSTEM Cached : 0
mid05tor12cn05: SYSTEM FieldReplaceable : 0
mid05tor12cn05: SYSTEM Manufacturer :
mid05tor12cn05: SYSTEM Model : 8335-GTC........
mid05tor12cn05: SYSTEM PartNumber :
mid05tor12cn05: SYSTEM Present : 1
mid05tor12cn05: SYSTEM PrettyName :
...
```

### CODE REVIEW

**Comment by robin2008** [comment]
And check if it will miss the zero there
```
use xCAT::Utils qw/natural_sort_cmp/;
sub natural_sorter {
    natural_sort_cmp( $a, $b );
}
my @outputs=(
    '781A09A',
    'node100',
    'node10',
    'node2',
    'node1',
    'node1-r2-cn03',
    'node1-r2-cn01',
);
print join( "\n", sort natural_sorter @outputs) . "\n";
```
output as below, there is no missing:
```
perl test.pl
781A09A
node1
node1-r2-cn01
node1-r2-cn03
node2
node10
node100
```

**Comment by robin2008** [comment]
Performance test with 5k elements.
```
#!/usr/bin/perl
BEGIN { $::XCATROOT = $ENV{'XCATROOT'} ? $ENV{'XCATROOT'} : -d '/opt/xcat' ? '/opt/xcat' : '/usr'; }
use lib "$::XCATROOT/lib/perl";
use Time::HiRes qw(sleep time);
use xCAT::Utils qw/natural_sort_cmp/;
sub natural_sorter {
    natural_sort_cmp( $a, $b );
}
my @outputs=`nodels`;
my $start=time();
#print join( "\n", sort natural_sorter @outputs) . "\n";
my @abc=sort natural_sorter @outputs;
printf("Elapsed time:%.6f\n", time()-$start);
#print join( "\n", @abc) . "\n";

my $start=time();
my @sorted_output = grep {s/(^|\D)0+(\d)/$1$2/g,1} sort
    grep {s/(\d+)/sprintf"%06.6d",$1/ge,1} @outputs;
printf("Elapsed time of previous sorting:%.6f\n", time()-$start);
```
Looks good, a bit slow than before.
```
perl test.pl
Elapsed time:0.088773
Elapsed time of previous sorting:0.028390
```

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -1887,12 +1892,13 @@ sub rinv_response {
...
         }
     }
-    # If sorted array has any contents, sort it and print it
+    # If sorted array has any contents, sort it naturally and print it
     if (scalar @sorted_output > 0) {
         # sort alpha, then numeric 
-        my @sorted_output = grep {s/(^|\D)0+(\d)/$1$2/g,1} sort 
-            grep {s/(\d+)/sprintf"%06.6d",$1/ge,1} @sorted_output;
-        foreach (@sorted_output) { 
+        #my @sorted_output = grep {s/(^|\D)0+(\d)/$1$2/g,1} sort
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: I don't think we need to keep this code around, we can delete it..  the 2 lines here..

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 2]
```diff
@@ -2390,9 +2397,10 @@ sub rvitals_response {
     # If sorted array has any contents, sort it and print it
     if (scalar @sorted_output > 0) {
         # Sort the output, alpha, then numeric
-        my @sorted_output = grep {s/(^|\D)0+(\d)/$1$2/g,1} sort 
-            grep {s/(\d+)/sprintf"%06.6d",$1/ge,1} @sorted_output;
-        xCAT::SvrUtils::sendmsg("$_", $callback, $node) foreach (@sorted_output);
+        #my @sorted_output = grep {s/(^|\D)0+(\d)/$1$2/g,1} sort
```
Comment (whowutwut) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: I don't think we need to keep this code around, we can delete it..  the 2 lines here..

**Comment by robin2008** [comment]
Updated.
1, remove the comments for previous codes
2, As the object is not used, not using an new array variable to increase the performance

**Comment by immarvin** [comment]
agree to merge

**Comment by samveen** [comment]
A little late, but there is no need for the `natural_sorter` function as a wrapper for `natural_sort_cmp`. Just adds another layer of indirection without any value.

**Pull request merged**